### PR TITLE
Fix a regession where an invalid path pattern is generated for gRPC trancoding

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

PR #5679 allows groups name in a regex path pattern to start with `@`. However, group names of a regular expression must start with a letter. https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#groupname

Modifications:

- Use `p\d` as a group name for `PathMappingType.REGEX`
  - `@p\d` will stay unchanged for `PathMappingType.PARAMETERIZED`

Result:

Fix a regression caused by #5679

### CODE REVIEW

File: grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingPathParser.java [hunk 1]
```diff
@@ -389,7 +389,7 @@ public String pathVariable(PathMappingType type) {
                 if (parentFieldPath != null) {
                     return parentFieldPath;
                 } else {
-                    return "@p" + StringUtil.toString(pathVarIndex);
+                        return 'P' + StringUtil.toString(pathVarIndex);
```
Comment (jrhee17) [inline grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingPathParser.java:392]:
```suggestion
                    return 'P' + StringUtil.toString(pathVarIndex);
```

**Review by jrhee17 — COMMENTED** [review]
In general, what do you think of always using the `P\\d` pattern for `REGEX` case so that underscored variable names are also supported?

**Comment by ikhoon** [comment]
> In general, what do you think of always using the `P\\d` pattern for `REGEX` case so that underscored variable names are also supported?

Agreed. I think we are on the same page. Addressed at [`16df88b` (#5729)](https://github.com/line/armeria/pull/5729/commits/16df88b055d757b708c75af329ce8ec806943b05)

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Review by minwoox — APPROVED** [review]
Thanks!

**Comment by minwoox** [comment]
OOps we should fix this test:
```
HttpJsonTranscodingTest > conflictingParamNamesReturnsCorrectly() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: 200
     but was: 404
```

**Comment by minwoox** [comment]
@ikhoon 👍 👍 👍

**Pull request merged**